### ISSUE TITLE
Adding Delay to Display Traffic Graph on Weathermap Links

### ISSUE THREAD

**Issue by jackfish0405** [thurban/plugin_weathermap#149]
Version 1.0 or later， On the weathermap page, when hovering over a link, the corresponding traffic graph immediately appears. However, the speed is too fast, making it difficult to accurately interact with densely packed links as the graph appears almost instantly upon mouseover. I would like to modify it so that the traffic graph only appears 1-2 seconds after the mouse hovers over the corresponding link. Which parameters should be modified to achieve this?

**Comment by TheWitness** [comment]
If you search for "hover" in the source, the numeric value there is in milliseconds.

It's a simple fix, but I'm busy now.  See what you can do.  It's using jQyeryUI dialog.

**Comment by TheWitness** [comment]
Okay, look in weather-map.js.  You will find 200 and 1000 values.  The 200 is the entry delay, the 1000 is the time it takes before starting to fade out the graph.

You will have to force reload the cached script if you are using Chrome.  Clearing the site browser cache should help there.  There are plugins to help with that if you use Firefox for example.

**Comment by jackfish0405** [comment]
Thanks for your help. I modified the content in the weathermap.js file, 
wmHoverTimeout = setTimeout(adjustTooltipWindow, 200); 
to wmHoverTimeout = setTimeout(adjustTooltipWindow, 2000); 
and cleared Chrome's cache, but it seems to have no effect. Is there anything else I need? operate?

**Comment by TheWitness** [comment]
You didn't modify the fade attribute that was so important one.

**Comment by jackfish0405** [comment]
I modified it again, and now I find that it sometimes takes effect and sometimes does not take effect when the mouse is moved over it.

            wmHoverTimeout = setTimeout(adjustTooltipWindow, 3000);
        },
        close: function(event, ui) {
            ui.tooltip.hover(
                function() {
                    $(this).stop(true).fadeTo(3000, 1);
                },
                function() {
                    $(this).fadeOut(3000, function() {
                        $(this).remove();
                    });
                }
            );
        },
![8](https://github.com/Cacti/plugin_weathermap/assets/53333879/e70feb2d-25ed-4a5a-9d64-9659c28a8821)

**Comment by TheWitness** [comment]
I have some other examples floating around, but it'll likely take me a few more days to dig them out as it's the weekend you know.

**Comment by channdy** [comment]
Hi @TheWitness, I am encountered the same issue. I want to have a small delay on mouse hover for each link and I have modified as follow but seems not take effect. Please help me on this. Thanks

wmHoverTimeout = setTimeout(adjustTooltipWindow, 5000);

**Comment by MatKra84** [comment]
This change in the file weathermap.js: works for me:

$(function () {
    var hoverTimeout;
    var $tooltipDiv;

    $('map area').on('mouseenter', function (e) {
        var $this = $(this);
        var caption = $this.attr('data-caption');
        var hover = $this.attr('data-hover');

        // 
        hoverTimeout = setTimeout(function () {
            $tooltipDiv = $('<div id="wm_custom_hover" class="ui-tooltip cactiTable"></div>').appendTo('body');

            var content = `
                <div><div id="wm_hover_child" class="cactiTableTitleRow">${caption}</div></div>
                <table class="cactiTable"><tr><td class="wmcontent">${hover}</td></tr></table>
            `;

            $tooltipDiv
                .html(content)
                .css({
                    position: 'absolute',
                    top: e.pageY,
                    left: e.pageX + 15,
                    'min-width': '200px',
                    'max-width': '400px',
                    'z-index': 10000,
                    'background': '#fff',
                    'border': '1px solid #888',
                    'padding': '4px',
                    'box-shadow': '2px 2px 8px rgba(0,0,0,0.3)',
                    'pointer-events': 'none'
                })
                .fadeIn(200);
        }, 500); // <-  Delay
    });

    $('map area').on('mouseleave', function () {
        clearTimeout(hoverTimeout);
        if ($tooltipDiv) {
            $tooltipDiv.fadeOut(200, function () {
                $(this).remove();
            });
        }
    });
});

function adjustTooltipWindow() {
        var maxWidth = $(tooltipObject).find('img').width();
        //console.log(maxWidth);
        tooltipObject.css('max-width', maxWidth+'px');
        tooltipObject.css('max-height', '800px');
        tooltipObject.css('overflow-y', 'visible');
}

**Comment by xiconfjs** [comment]
This is my fix @ weathermap v1.3

```javascript
--- /tmp/weathermap_orig.js	2025-09-01 11:25:14.998506300 +0000
+++ /usr/share/cacti/site/plugins/weathermap/js/weathermap.js	2025-09-01 11:24:46.745652086 +0000
@@ -4,6 +4,7 @@
 $(function() {
 	$('map').tooltip({
 		items: 'area',
+		show: { delay: 500 },
 		open: function(event, ui) {
 			if (typeof(event.originalEvent) == 'undefined') {
 				return false;
@@ -18,7 +19,7 @@
 			ui.tooltip.css('overflow-y', 'visible');
 
 			ui.tooltip.position({
-				my: 'left top',
+				//my: 'left top',
 				at: 'right+15 center',
 				of: event
 			});

```

**Comment by TheWitness** [comment]
@xiconfjs @channdy @jackfish0405 and @MatKra84, I'm working on this plugin this week.  I'm trying to fix somethings up.  If you guys have a test environment where you can test the proposed changes, I would appreciate it.

**Comment by TheWitness** [comment]
I'm going to mark this one resolved and close it guys.  Give 1.4 a spin once it's released, and if this issue persists, let me know via a new ticket.

**Comment by xiconfjs** [comment]
> [@xiconfjs](https://github.com/xiconfjs) [@channdy](https://github.com/channdy) [@jackfish0405](https://github.com/jackfish0405) and [@MatKra84](https://github.com/MatKra84), I'm working on this plugin this week. I'm trying to fix somethings up. If you guys have a test environment where you can test the proposed changes, I would appreciate it.

Hey @TheWitness, happy to test your changes.

**Comment by TheWitness** [comment]
Jump in then and thanks!  I pushed 1.4, but there are more changes coming.  This was a catch-up release.

**Comment by xiconfjs** [comment]
> Jump in then and thanks! I pushed 1.4, but there are more changes coming. This was a catch-up release.

first, thank you for your work, very much appreciated. Second, I added the following line, to move the tooltip a bit to the right so it is easier to click the link/node below the tooltip:

```javascript
// @ /usr/share/cacti/site/plugins/weathermap/js/weathermap.js

function adjustTooltipWindow() {
        $('.ui-tooltip').find('img').css('max-width', '100%');
+       $('.ui-tooltip').css('transform', 'translateX(+20px)');
        $('.wmcontent').show();
}
```

**Comment by TheWitness** [comment]
Looks good to me.  Let me squeeze that into the next batch of changes.
